pkg/reconciler/instances/eventing/preaction/removenatsoperatorstep.go

```diff
@@ -0,0 +1,109 @@
+package preaction
+
+import (
+	"github.com/kyma-incubator/reconciler/pkg/reconciler/chart"
+	"github.com/kyma-incubator/reconciler/pkg/reconciler/instances/eventing/log"
+	"github.com/kyma-incubator/reconciler/pkg/reconciler/kubernetes"
+	"github.com/kyma-incubator/reconciler/pkg/reconciler/service"
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"strings"
+)
+
+const (
+	removeNatsOperatorStepName = "removeNatsOperator"
+	natsOperatorLastVersion    = "1.24.7"
+	natsSubChartPath           = "eventing/charts/nats"
+	eventingNats               = "eventing-nats"
+	oldConfigValue             = "global.image.repository"
+	newConfigValue             = "eu.gcr.io/kyma-project"
+)
+
+var (
+	natsOperatorCRDsToDelete = []string{"natsclusters.nats.io", "natsserviceroles.nats.io"}
+)
+
+type kubeClientProvider func(context *service.ActionContext, logger *zap.SugaredLogger) (kubernetes.Client, error)
+
+type removeNatsOperatorStep struct {
+	kubeClientProvider
+}
+
+func newRemoveNatsOperatorStep() *removeNatsOperatorStep {
+	return &removeNatsOperatorStep{
+		kubeClientProvider: defaultKubeClientProvider,
+	}
+}
+
+func defaultKubeClientProvider(context *service.ActionContext, logger *zap.SugaredLogger) (kubernetes.Client, error) {
+	kubeClient, err := kubernetes.NewKubernetesClient(context.Task.Kubeconfig, logger, &kubernetes.Config{
+		ProgressInterval: progressTrackerInterval,
+		ProgressTimeout:  progressTrackerTimeout,
+	})
+	if err != nil {
+		return nil, err
+	}
+	return kubeClient, nil
+}
+
+func (r *removeNatsOperatorStep) Execute(context *service.ActionContext, logger *zap.SugaredLogger) error {
+	kubeClient, err := r.kubeClientProvider(context, logger)
+	if err != nil {
+		return err
+	}
+
+	// decorate logger
+	logger = logger.With(log.KeyStep, removeNatsOperatorStepName)
+
+	err = r.removeNatsOperatorResources(context, kubeClient, logger)
+	if err != nil {
+		return err
+	}
+
+	err = r.removeNatsOperatorCRDs(kubeClient, logger)
+	return err
+}
+
+func (r *removeNatsOperatorStep) removeNatsOperatorResources(context *service.ActionContext, kubeClient kubernetes.Client, logger *zap.SugaredLogger) error {
+	// get charts from the version 1.2.x, where the NATS-operator resources still exist
+	comp := chart.NewComponentBuilder(natsOperatorLastVersion, natsSubChartPath).
+		WithConfiguration(map[string]interface{}{
+			oldConfigValue: newConfigValue, // replace the missing global value, as we are rendering on the subchart level
+		}).
+		WithNamespace(namespace).
+		Build()
```

Comment:
Please extract this code into a function so we can use it in the test assertions.